pkg/agent/openflow/pipeline_windows.go

```diff
@@ -140,3 +140,96 @@ func (c *client) l3FwdFlowToRemoteViaRouting(localGatewayMAC net.HardwareAddr, r
+	var flows []binding.Flow
+
+	connectionTrackCommitTable := c.pipeline[conntrackCommitTable]
+	if c.enableProxy {
+		// These 2 tables are for traffic SNAT of ClusterIP Service from Windows host.
+		serviceConnectionTrackTable := c.pipeline[serviceConntrackTable]
+		serviceConnectionTrackCommitTable := c.pipeline[serviceConntrackCommitTable]
+
+		flows = append(flows,
+			c.pipeline[spoofGuardTable].BuildFlow(priorityHigh).MatchProtocol(binding.ProtocolIP).
+				MatchInPort(config.HostGatewayOFPort).MatchDstIP(config.ServiceGWHairpinIPv4).
+				MatchDstMAC(globalVirtualMAC).
+				Action().GotoTable(serviceConntrackTable).
+				Cookie(c.cookieAllocator.Request(category).Raw()).
+				Done(),
+			serviceConnectionTrackTable.BuildFlow(priorityNormal).MatchProtocol(binding.ProtocolIP).
+				Action().CT(false, serviceConnectionTrackTable.GetNext(), SNATCtZone).NAT().CTDone().
+				Cookie(c.cookieAllocator.Request(category).Raw()).
+				Done(),
+			connectionTrackCommitTable.BuildFlow(priorityHigh).MatchProtocol(binding.ProtocolIP).
+				Cookie(c.cookieAllocator.Request(category).Raw()).
+				MatchCTMark(ServiceCTMark).
+				MatchRegMark(FromGatewayRegMark).
+				Action().GotoTable(serviceConntrackCommitTable).
+				Cookie(c.cookieAllocator.Request(category).Raw()).
+				Done(),
+			connectionTrackCommitTable.BuildFlow(priorityNormal).MatchProtocol(binding.ProtocolIP).
+				MatchCTMark(ServiceCTMark).MatchCTStateNew(false).MatchCTStateTrk(true).
+				Action().GotoTable(hairpinSNATTable).
+				Cookie(c.cookieAllocator.Request(category).Raw()).
+				Done(),
+			serviceConnectionTrackCommitTable.BuildFlow(priorityHigh).MatchProtocol(binding.ProtocolIP).
+				Cookie(c.cookieAllocator.Request(category).Raw()).
+				MatchInPort(config.HostGatewayOFPort).
+				MatchCTStateNew(true).
+				MatchCTStateTrk(true).
+				MatchCTStateDNAT(true).
+				MatchDstMAC(virtualSvcMAC).
+				Action().LoadRegMark(HairpinRegMark).
+				Action().CT(true, serviceConnectionTrackCommitTable.GetNext(), SNATCtZone).
+				SNAT(&binding.IPRange{StartIP: config.ServiceGWHairpinIPv4, EndIP: config.ServiceGWHairpinIPv4}, nil).
+				CTDone().
+				Done(),
+			serviceConnectionTrackCommitTable.BuildFlow(priorityNormal).MatchProtocol(binding.ProtocolIP).
+				Cookie(c.cookieAllocator.Request(category).Raw()).
+				MatchInPort(config.HostGatewayOFPort).
+				MatchCTStateNew(false).
+				MatchCTStateTrk(true).
+				Action().CT(false, serviceConnectionTrackCommitTable.GetNext(), SNATCtZone).NAT().
+				CTDone().
+				Done(),
+		)
+	}
+	return flows
+}
+
+func (c *client) l3FwdFlowClusterIPServices(localGatewayMAC net.HardwareAddr, category cookie.Category) []binding.Flow {
+	var flows []binding.Flow
+	l3FwdTable := c.pipeline[l3ForwardingTable]
+	flows = append(flows, []binding.Flow{
+		l3FwdTable.BuildFlow(priorityLow).MatchProtocol(binding.ProtocolIP).
+			Cookie(c.cookieAllocator.Request(category).Raw()).
+			MatchCTStateTrk(true).
+			MatchCTMark(ServiceCTMark).
```

Comment:
If my understanding is correct, you want to rewrite the dst MAC as antrea-gw0's MAC if it is performed DNAT. Is it possible to be overlapped with the one to resubmit packets to snatTable for all packets sent out from the local Pods, which one also has a low priority?

Reply:
Updated.